enterprise/backend/test/metabase_enterprise/cache/cache_test.clj

```diff
@@ -111,3 +112,80 @@
+                                                  :model_id       (:id dash)
+                                                  :strategy       "schedule"
+                                                  :config         {:schedule "0 * * * * ?"}
+                                                  :invalidated_at (t/offset-date-time)}]
+
+          (is (=? {:data [{:model "database" :model_id (mt/id)}]}
+                  (mt/user-http-request :crowberto :get 200 "cache/"
+                                        :model "database")))
+
+          (testing "making a query will cache it"
+            (is (=? {:cached false :data some?}
+                    (mt/user-http-request :crowberto :post 202 (format "card/%d/query" (:id card)))))
+            (is (=? {:cached true :data some?}
+                    (mt/user-http-request :crowberto :post 202 (format "card/%d/query" (:id card)))))
+            (is (=? {:cached true :data some?}
+                    (mt/user-http-request :crowberto :post 202 (format "card/%d/query" (:id card2))))))
+
+          (testing "invalidation drops cache only for affected card"
+            (is (=? {:count 1}
+                    (mt/user-http-request :crowberto :post 200 "cache/invalidate"
+                                          :question (:id card2) :include :overrides)))
+            (is (=? {:cached true :data some?}
+                    (mt/user-http-request :crowberto :post 202 (format "card/%d/query" (:id card)))))
+            (is (=? {:cached false :data some?}
+                    (mt/user-http-request :crowberto :post 202 (format "card/%d/query" (:id card2))))))
+
+          (testing "but invalidating a whole config drops cache for all affected cards"
+            (is (=? {:count 1}
+                    (mt/user-http-request :crowberto :post 200 "cache/invalidate"
+                                          :database (mt/id))))
+            (is (=? {:cached false :data some?}
+                    (mt/user-http-request :crowberto :post 202 (format "card/%d/query" (:id card)))))
+            ;; we need to drop it again so that new cache is not being used
+            (is (=? {:count 1}
+                    (mt/user-http-request :crowberto :post 200 "cache/invalidate"
+                                          :database (mt/id))))
+            (is (=? {:cached false :data some?}
+                    (mt/user-http-request :crowberto :post 202 (format "card/%d/query" (:id card2))))))
+
+          (testing "when invalidating database config with no overrides, dashboard-related queries will still have it"
+            (is (=? {:count 1}
+                    (mt/user-http-request :crowberto :post 200 "cache/invalidate"
+                                          :database (mt/id))))
+            (is (=? {:cached true :data some?}
+                    (mt/user-http-request :crowberto :post 202 (format "card/%d/query" (:id card))
+                                          {:dashboard_id (:id dash)}))))
```

Comment:
This is something I'm not sure is exactly _nice_, feels like a weird edge case, but agreed on that behavior leaves us with a simple to understand solution - while trying to improve on the situation (which I did in my previous attempt) makes for lots of code having it's own weird edge cases.

So I think this is acceptable.